app/main/models/models.py

```diff
@@ -83,6 +129,7 @@ class Source(BaseModel):
     # The project that holds this source
     project_id = db.Column(db.Integer, db.ForeignKey('projects.id'),
                            nullable=False)
+    cluster_id = db.Column(db.Integer, db.ForeignKey('clusters.id'))
```

Comment:
Refer to my comment on `Cluster`.  Would a source have both a `project_id` and a `cluster_id` if it's inside a cluster?